src/main/java/com/searchspring/nextopia/Converter.java

```diff
@@ -53,8 +65,37 @@ private String getTotalResults(SearchspringResponse response) {
         return "0";
     }
 
-    public String convertNextopiaQueryUrl(String nextopiaQueryUrl) {
-        return "newUrl";
+    public String convertNextopiaQueryUrl(String nextopiaQueryUrl) throws URISyntaxException {
+        URI uri = new URI(nextopiaQueryUrl);
+        Map<String, String> queryMap = parseQueryString(uri.getQuery());
+        StringBuilder sb = new StringBuilder("http://").append(siteId).append(".a.searchspring.io/api?siteId=").append(siteId);
```

Comment:
I disagree that this is magic.   Magic is where you can't see what the code is doing, like AOP and reflection.  Though I do think it could be clearer - I think extracting it to a method would make it clearer.  like `prepend